src/language/ast.ts

```diff
@@ -358,10 +375,36 @@ export interface FieldNode {
   readonly alias?: NameNode;
   readonly name: NameNode;
   readonly arguments?: ReadonlyArray<ArgumentNode>;
+  // Note: Client Controlled Nullability is experimental
+  // and may be changed or removed in the future.
+  readonly required?: ListNullabilityNode | NullabilityDesignatorNode;
   readonly directives?: ReadonlyArray<DirectiveNode>;
   readonly selectionSet?: SelectionSetNode;
 }
 
+export interface RequiredDesignatorNode {
```

Comment:
We might want to change the name of these nodes to match the new null propagation behavior. I'll have to come back to this.